gdk-pixbuf/src/subclass/pixbuf_loader.rs

```diff
@@ -0,0 +1,158 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+// rustdoc-stripper-ignore-next
+//! Traits intended for subclassing [`PixbufLoader`](crate::PixbufLoader).
+
+use crate::PixbufLoader;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+pub trait PixbufLoaderImpl: ObjectImpl {
+    fn size_prepared(&self, width: i32, height: i32) {
+        self.parent_size_prepared(width, height)
+    }
+
+    fn area_prepared(&self) {
+        self.parent_area_prepared()
+    }
+
+    fn area_updated(&self, x: i32, y: i32, width: i32, height: i32) {
+        self.parent_area_updated(x, y, width, height)
+    }
+
+    fn closed(&self) {
+        self.parent_closed()
+    }
+}
+
+pub trait PixbufLoaderImplExt: ObjectSubclass {
+    fn parent_size_prepared(&self, width: i32, height: i32);
+    fn parent_area_prepared(&self);
+    fn parent_area_updated(&self, x: i32, y: i32, width: i32, height: i32);
+    fn parent_closed(&self);
+}
+
+impl<T: PixbufLoaderImpl> PixbufLoaderImplExt for T {
+    fn parent_size_prepared(&self, width: i32, height: i32) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GdkPixbufLoaderClass;
+            let f = (*parent_class)
+                .size_prepared
+                .expect("No parent class implementation for \"size_prepared\"");
```

Comment:
All the vfuncs here are signal class handlers. So if they don't exist (which they don't by default), you would simply do nothing at all here and return quitely instead of panicking